### DISCUSSION TITLE
Booting from vhost block device

### DISCUSSION THREAD

**Discussion by maburlik** [cloud-hypervisor/cloud-hypervisor#3695] [Q&A]
Is it possible today to boot from a vhost_user block device created using SPDK? If it is, would this be possible using the standard rust hypervisor firmware & Ubuntu cloud image?

i.e.
Using the Ubuntu focal-server-cloudimg-amd64 image, somehow write it to the vhost device, and:

sudo ./cloud-hypervisor \
        --cpus boot=1 \
        --memory size=1024M,hugepages=on,shared=true \
        --kernel /tmp/hypervisor-fw \
        --cmdline "console=ttyS0 root=/dev/vda1 rw iommu=off" \
        --disk vhost_user=true,socket=/var/tmp/vhost.1,num_queues=1,queue_size=128 \
        --console off \
        --serial tty \
        --rng

Thanks.

**Comment by rbradford** [comment]
Yes, it should work. How is it failing for you?

**Reply by maburlik** [reply]
> Yes, it should work. How is it failing for you?

I am trying to figure out how to write the image to the block device.

**Reply by rbradford** [reply]
Use something like `./scripts/rpc.py bdev_aio_create ~/workloads/focal-server-cloudimg-amd64.raw test 512` if you're using SPDK.
